src/tests/ftest/telemetry/dkey_akey_enum_punch.py

```diff
@@ -0,0 +1,371 @@
+        ]
+        pool_out = self.telemetry.get_pool_metrics(
+            specific_metrics=specific_metrics)
+
+        # Verify dkey_enum total is 100.
+        dkey_enum_total = self.sum_values(metric_out=pool_out[pool_dkey_enum])
+        if dkey_enum_total != 100:
+            msg = "dkey enum total is not 100! Actual = {}".format(
+                dkey_enum_total)
+            self.errors.append(msg)
+
+        # Verify akey_enum total is 200.
+        akey_enum_total = self.sum_values(metric_out=pool_out[pool_akey_enum])
+        if akey_enum_total != 200:
+            msg = "akey enum total is not 200! Actual = {}".format(
+                akey_enum_total)
+            self.errors.append(msg)
+
+        # Verify dkey_punch total is 200.
+        dkey_punch_total = self.sum_values(metric_out=pool_out[pool_dkey_punch])
+        if dkey_punch_total != 200:
+            msg = "dkey punch total is not 200! Actual = {}".format(
+                dkey_punch_total)
+            self.errors.append(msg)
+
+        # Verify akey_punch total is 200.
+        akey_punch_total = self.sum_values(metric_out=pool_out[pool_akey_punch])
+        if akey_punch_total != 200:
+            msg = "akey punch total is not 200! Actual = {}".format(
+                akey_punch_total)
+            self.errors.append(msg)
+
+        ### Verify active and latency; metrics 5 to 8. ###
+        # Verify dkey enum active.
+        prefix = "engine_io_ops_dkey_enum_active_"
+        dkey_active_metrics = self.get_min_max_mean_stddev(
+            prefix=prefix, total_targets=self.total_targets,
+            target_per_rank=self.target_per_rank)
+
+        self.errors.extend(
+            self.verify_stats(
+                enum_metrics=dkey_active_metrics, metric_prefix=prefix,
+                test_latency=False))
+
+        # Verify akey enum active.
+        prefix = "engine_io_ops_akey_enum_active_"
+        akey_active_metrics = self.get_min_max_mean_stddev(
+            prefix=prefix, total_targets=self.total_targets,
+            target_per_rank=self.target_per_rank)
+
+        self.errors.extend(
+            self.verify_stats(
+                enum_metrics=akey_active_metrics, metric_prefix=prefix,
+                test_latency=False))
+
+        # Verify dkey enum latency.
+        prefix = "engine_io_ops_dkey_enum_latency_"
+        dkey_latency_metrics = self.get_min_max_mean_stddev(
+            prefix=prefix, total_targets=self.total_targets,
+            target_per_rank=self.target_per_rank)
+
+        self.errors.extend(
+            self.verify_stats(
+                enum_metrics=dkey_latency_metrics, metric_prefix=prefix,
+                test_latency=True))
+
+        # Verify akey enum latency.
+        prefix = "engine_io_ops_akey_enum_latency_"
+        akey_latency_metrics = self.get_min_max_mean_stddev(
+            prefix=prefix, total_targets=self.total_targets,
+            target_per_rank=self.target_per_rank)
```

Comment:
Minor: I think a helper function would reduce this code repletion (maybe even just define in the scope of this function):
```
def helper(prefix):
        metrics = self.get_min_max_mean_stddev(
            prefix=prefix, total_targets=self.total_targets,
            target_per_rank=self.target_per_rank)

        self.errors.extend(
            self.verify_stats(
                enum_metrics=metrics, metric_prefix=prefix,
                test_latency=True))

helper("engine_io_ops_dkey_enum_active_")
helper("engine_io_ops_akey_enum_active_""
helper("engine_io_ops_dkey_enum_latency_")
helper(""engine_io_ops_akey_enum_latency_")
```

Reply:
Done. Thanks!